massa-proto/proto/massa/api/v1/execution.proto

```diff
@@ -67,16 +70,226 @@ message ScExecutionEventContext {
+  AsyncPoolChangeValue value = 2;
+}
+
+// AsyncPoolChangeValue
+message AsyncPoolChangeValue {
+  // The type of the change
+  AsyncPoolChangeType type = 1;
+  // AsyncPool message
+  AsyncMessage async_message = 2;
+}
+
+// AsyncPoolChangeType type enum
+enum AsyncPoolChangeType {
+  ASYNC_POOL_CHANGE_TYPE_UNSPECIFIED = 0; // Default enum value
+  ASYNC_POOL_CHANGE_TYPE_ADD = 1; // Add type
+  ASYNC_POOL_CHANGE_TYPE_ACTIVATE = 2; // Activate only type
+  ASYNC_POOL_CHANGE_TYPE_DELETE = 3; // Delete only type
+}
+
+// Asynchronous smart contract message
+message AsyncMessage {
+  // Slot at which the message was emitted
+  Slot emission_slot = 1;
+
+  // Index of the emitted message within the `emission_slot`.
+  // This is used for disambiguate the emission of multiple messages at the same slot.
+  fixed64 emission_index = 2;
+
+  // The address that sent the message
+  string sender = 3;
+
+  // The address towards which the message is being sent
+  string destination = 4;
+
+  // the handler function name within the destination address' bytecode
+  string handler = 5;
+
+  // Maximum gas to use when processing the message
+  fixed64 max_gas = 6;
+
+  // Fee paid by the sender when the message is processed.
+  fixed64 fee = 7;
+
+  // Coins sent from the sender to the target address of the message.
+  // Those coins are spent by the sender address when the message is sent,
+  // and credited to the destination address when receiving the message.
+  // In case of failure or discard, those coins are reimbursed to the sender.
+  fixed64 coins = 8;
+
+  // Slot at which the message starts being valid (bound included in the validity range)
+  Slot validity_start = 9;
+
+  // Slot at which the message stops being valid (bound not included in the validity range)
+  Slot validity_end = 10;
+
+  // Raw payload data of the message
+  bytes data = 11;
+
+  // Trigger that define whenever a message can be executed (optional)
+  AsyncMessageTrigger trigger = 12;
+
+  // Boolean that determine if the message can be executed. For messages without filter this boolean is always true.
+  // For messages with filter, this boolean is true if the filter has been matched between `validity_start` and current slot.
+  bool can_be_executed = 13;
+
+  // Hash of the message
+  string hash = 14;
+}
+
+// Structure defining a trigger for an asynchronous message
+message AsyncMessageTrigger {
+    // Filter on the address
+    string address = 1;
+    // Filter on the datastore key (optional)
+    optional bytes datastore_key = 2;
+}
+
+// Ledger change
+message LedgerChanges {
+  // Ledger change entry
+  repeated LedgerChangeEntry entries = 2;
+}
+
+// LedgerChangeEntry
+message LedgerChangeEntry {
+  // Address
+  string address = 1;
+  // Ledger message
+  LedgerChangeValue value = 2;
+}
+
+// LedgerChangeValue
+message LedgerChangeValue {
+  // The type of the change
+  LedgerChangeType type = 1;
+  // Created ledger entry
+  LedgerEntry created_entry = 2;
```

Comment:
This is either this one or the update one. Is that possible to define this "or" in grpc?

Reply:
Sorry, I was quiet tired when I implemented this. Of course we could make It inn gRPC

Reply:
done